generators/java/sdk/src/main/java/com/fern/java/client/generators/endpoint/AbstractEndpointWriter.java

```diff
@@ -714,6 +881,229 @@ private boolean isAliasContainer(com.fern.irV42.model.types.TypeReference respon
         }
     }
 
+    private SnippetAndResultType getNestedPropertySnippet(
+            Optional<List<Name>> propertyPath,
+            ObjectProperty objectProperty,
+            com.fern.irV42.model.types.TypeReference typeReference) {
+        ArrayList<Name> fullPropertyPath = propertyPath.map(ArrayList::new).orElse(new ArrayList<>());
+        fullPropertyPath.add(objectProperty.getName().getName());
+        GetSnippetOutput getSnippetOutput = typeReference.visit(new NestedPropertySnippetGenerator(
+                typeReference, fullPropertyPath, false, false, Optional.empty(), Optional.empty()));
+        Builder codeBlockBuilder = CodeBlock.builder();
+        getSnippetOutput.code.forEach(codeBlockBuilder::add);
+        return new SnippetAndResultType(
+                getSnippetOutput.typeReference,
+                clientGeneratorContext.getPoetTypeNameMapper().convertToTypeName(true, getSnippetOutput.typeReference),
+                codeBlockBuilder.build());
+    }
+
+    private class SnippetAndResultType {
+        private final com.fern.irV42.model.types.TypeReference typeReference;
+        private final TypeName typeName;
+        private final CodeBlock codeBlock;
+
+        private SnippetAndResultType(
+                com.fern.irV42.model.types.TypeReference typeReference, TypeName typeName, CodeBlock codeBlock) {
+            this.typeReference = typeReference;
+            this.typeName = typeName;
+            this.codeBlock = codeBlock;
+        }
+    }
+
+    private class NestedPropertySnippetGenerator
+            implements com.fern.irV42.model.types.TypeReference.Visitor<GetSnippetOutput> {
```

Comment:
This turned out very not elegant -- there may be a better way, but I couldn't figure it out